testing/src/main/java/net/consensys/linea/testing/ReplayExecutionEnvironment.java

```diff
@@ -0,0 +1,286 @@
+      final BlockHeader header = blockSnapshot.header().toBlockHeader();
+
+      final BlockBody body =
+          new BlockBody(
+              blockSnapshot.txs().stream().map(TransactionSnapshot::toTransaction).toList(),
+              new ArrayList<>());
+      tracer.traceStartBlock(header, body);
+
+      for (TransactionSnapshot txs : blockSnapshot.txs()) {
+        final Transaction tx = txs.toTransaction();
+        final WorldUpdater updater = world.updater();
+        // Process transaction leading to expected outcome
+        final TransactionProcessingResult outcome =
+            transactionProcessor.processTransaction(
+                updater,
+                (ProcessableBlockHeader) header,
+                tx,
+                header.getCoinbase(),
+                buildOperationTracer(tx, txs.getOutcome()),
+                blockHashLookup,
+                false,
+                Wei.ZERO);
+        // Commit transaction
+        updater.commit();
+      }
+      tracer.traceEndBlock(header, body);
+    }
+    tracer.traceEndConflation(world.updater());
+  }
+
+  /**
+   * Construct transaction processor a given chain (e.g. 59144 for mainnet, 59141 for sepolia, etc).
+   *
+   * @return
+   */
+  private MainnetTransactionProcessor getMainnetTransactionProcessor(BigInteger chainId) {
+    EvmConfiguration evmConfig = EvmConfiguration.DEFAULT;
+    // Read genesis config for linea
+    GenesisConfigFile configFile =
+        GenesisConfigFile.fromSource(GenesisConfigFile.class.getResource("/linea.json"));
+    GenesisConfigOptions options = configFile.getConfigOptions();
+    BadBlockManager badBlockManager = new BadBlockManager();
+    // Create the schedule
+    ProtocolSchedule schedule =
+        MainnetProtocolSchedule.fromConfig(
+            options,
+            MiningParameters.MINING_DISABLED,
+            badBlockManager,
+            false,
+            new NoOpMetricsSystem());
+    // Create
+    ProtocolSpecBuilder builder =
+        new MainnetProtocolSpecFactory(
+                Optional.of(chainId),
+                true,
+                OptionalLong.empty(),
+                evmConfig,
+                MiningParameters.MINING_DISABLED,
+                false,
+                new NoOpMetricsSystem())
+            .londonDefinition(options); // .lineaOpCodesDefinition(options);
+    //
+    builder.privacyParameters(PrivacyParameters.DEFAULT);
+    builder.badBlocksManager(badBlockManager);
+    //
+    return builder.build(schedule).getTransactionProcessor();
+  }
+
+  public Hub getHub() {
+    return tracer.getHub();
+  }
+
+  /**
+   * Initialise a world updater given a conflation. Observe this can be applied to any WorldUpdater,
+   * such as SimpleWorld.
+   *
+   * @param world The world to be initialised.
+   * @param conflation The conflation from which to initialise.
+   */
+  private static void initWorld(WorldUpdater world, final ConflationSnapshot conflation) {
```

Comment:
Does a world that was thusly initialized know about precompiles ? I ask because in our old setup we had to invoke a special function to make the execution environment and the world aware of precompiles.